### DISCUSSION TITLE
How to skip type check for sfc files when the script is not ts(which does not have lang="ts")

### DISCUSSION THREAD

**Discussion by lovetingyuan** [johnsoncodehk/vue-tsc#46] [Q&A]
How to skip type check for sfc files when the script is not ts(which does not have lang="ts") ? 

I just want to perform type checking to ts file or vue sfc which only using typescript.

@johnsoncodehk

**Comment by johnsoncodehk** [comment]
Maybe we can have a tsconfig & IDE option like `volar.checkTemplate` to control template type-checking behavior.

**Never**: disable in all vue scripts
**WhenScriptLangIsTs**: enable if script language is `ts/tsx`
**Always**: enable in all vue scripts

**Reply by lovetingyuan** [reply]
OK, and additionally how about offering a cli argument like "vue-tsc --check-template langIsTs"?

**Reply by johnsoncodehk** [reply]
It may not possible, because it's out of proxy scope.

**Reply by johnsoncodehk** [reply]
Default behavior has be changed to **WhenScriptLangIsTs** in https://github.com/johnsoncodehk/volar/commit/51a6d4e2c8b12250e1c6e3f26fb09e74f61f7029.
